src/ex4/client/src/components/TodoInput/TodoInput.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from "react";
+import { apiSetTodo } from "../../services/TodoApiServies";
+import "./TodoInput.css";
+import Button from "../Button/Button";
+import PropTypes from "prop-types";
+
+const TodoInput = ({ todos, setTodos }) => {
+  const [inputValue, setInputValue] = useState("");
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setInputValue("");
+    apiSetTodo(inputValue).then((data) => {
+      Array.isArray(data)
+        ? setTodos((prevTodos) => [...prevTodos, ...data])
+        : setTodos((prevTodos) => [...prevTodos, data]);
```

Comment:
```suggestion
      Array.isArray(data)
        ? addTodos(data)
        : addTodos([data]);
```

Then you can define addTodo to c call ```setTodos((prevTodos) => [...prevTodos, ...data]) ```